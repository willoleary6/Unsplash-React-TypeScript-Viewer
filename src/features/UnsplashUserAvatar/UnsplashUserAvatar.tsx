import React from "react";

interface UnsplashUserAvatarProps {
    username: string;
    profileLink: string;
    profileImageUrl: string;
}

export const UnsplashUserAvatar = ({
    username,
    profileLink,
    profileImageUrl,
}: UnsplashUserAvatarProps): JSX.Element => {
    return (
        <div
            className=" bg-white px-4 pt-5 pb-4 sm:p-6 sm:pb-4 col-span-1 flex flex-col hover:cursor-pointer"
            onClick={() => window.open(profileLink, "_blank")}
        >
            <img
                alt="gallery"
                className="inline object-cover w-16 h-16 mr-2 rounded-full justify-center"
                src={profileImageUrl}
            />
            <p className="justify-center">{username}</p>
        </div>
    );
};
