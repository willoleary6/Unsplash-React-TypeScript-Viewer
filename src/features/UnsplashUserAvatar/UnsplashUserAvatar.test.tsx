import React from "react";
import { UnsplashUserAvatar } from "./UnsplashUserAvatar";

import Enzyme, { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

Enzyme.configure({ adapter: new Adapter() });

describe("SearchStatusBox render testing", () => {
    it("renders panel and takes snapshot to track jsx element changes, will also test for exceptions in element code", () => {
        const wrapper = shallow(
            <UnsplashUserAvatar username="test" profileLink="test" profileImageUrl="test" />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
