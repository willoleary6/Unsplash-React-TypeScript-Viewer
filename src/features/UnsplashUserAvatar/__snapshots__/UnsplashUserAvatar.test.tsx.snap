// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchStatusBox render testing renders panel and takes snapshot to track jsx element changes, will also test for exceptions in element code 1`] = `
<div
  className=" bg-white px-4 pt-5 pb-4 sm:p-6 sm:pb-4 col-span-1 flex flex-col hover:cursor-pointer"
  onClick={[Function]}
>
  <img
    alt="gallery"
    className="inline object-cover w-16 h-16 mr-2 rounded-full justify-center"
    src="test"
  />
  <p
    className="justify-center"
  >
    test
  </p>
</div>
`;
